src/lib/inputs/InputNumber.svelte

```diff
@@ -0,0 +1,397 @@
+        if (isLoading) {
+            containerClassList.push("loading");
+        } else if (isError) {
+            containerClassList.push("error");
+        }
+        containerClassString = containerClassList.join(" ");
+
+        readOnly = disabled || isLoading || isError;
+    }
+
+    $: inputStyle = `
+  background: ${disabled || isError || isLoading ? "#ccc" : background};
+  --color: ${color};
+  color: ${color};
+  ${style}
+`;
+
+    // Handle input event
+    const handleInput = (e: Event) => {
+        if (e.target instanceof HTMLInputElement) {
+            let inputValue = parseInt(e.target.value);
+
+            if (inputValue < min) {
+                inputValue = min;
+            } else if (inputValue > max) {
+                inputValue = max;
+            }
+
+            value = inputValue;
+            e.target.value = value.toString();
+            dispatch("input", e);
+        }
+    };
+
+    // Increment Value
+    const incrementValue = () => {
+        if (isLoading || isError || disabled) return;
+
+        if (Number.isNaN(value)) {
+            value = 0;
+        }
+        let newValue =
+            value + (typeof step === "number" ? step : parseInt(step));
+        if (newValue > max) {
+            newValue = max;
+        }
+        value = newValue;
+    };
+
+    // Decrement Value
+    const decrementValue = () => {
+        if (isLoading || isError || disabled) return;
+
+        if (Number.isNaN(value)) {
+            value = 0;
+        }
+        let newValue =
+            value - (typeof step === "number" ? step : parseInt(step));
+        if (newValue < min) {
+            newValue = min;
+        }
+        value = newValue;
+    };
+
+    // Handle Long Press
+    let incrementTimeout: any;
+    let decrementTimeout: any;
+    let incrementInterval: any;
+    let decrementInterval: any;
+
+    const startIncrementing = () => {
+        incrementTimeout = setTimeout(() => {
+            incrementInterval = setInterval(incrementValue, 100);
+        }, 500);
+    };
+
+    const stopIncrementing = () => {
+        clearTimeout(incrementTimeout);
+        clearInterval(incrementInterval);
+    };
+
+    const startDecrementing = () => {
+        decrementTimeout = setTimeout(() => {
+            decrementInterval = setInterval(decrementValue, 100);
+        }, 500);
+    };
+
+    const stopDecrementing = () => {
+        clearTimeout(decrementTimeout);
+        clearInterval(decrementInterval);
+    };
+</script>
+
+<div class={containerClassString}>
+    {#if label}
+        <label class="label" for={$$restProps.id} style="color: {labelColor}"
+            >{$$restProps.id}</label
+        >
+    {/if}
+
+    <input
+        {color}
+        {value}
+        {disabled}
+        {step}
+        {min}
+        {max}
+        readonly={readOnly}
+        {...$$restProps}
+        id={label ? $$restProps.id : undefined}
+        type="number"
+        style={inputStyle}
```

Comment:
I'm still getting used to the syntax on styling but between the various files ... can users adjust the width of this? Like what happens in the event the user wants to have something with a large number or long decimal

Reply:
This number input will adjust its width based on the min and max provided. OR the user can manually adjust the width using the style prop.